### ISSUE TITLE
Do common constraints apply for events in HTTP Baseline

### ISSUE THREAD

**Issue by egekorkan** [w3c/wot-profile#300]
There is security section in the common constraints section. My question is: What happens when there are events in a TD but that TD follows a profile that does not constrain events (like baseline)?
In other words, should an event in a baseline profile TD use API key security scheme?

**Comment by benfrancis** [comment]
> Do common constraints apply for events in HTTP Baseline

Yes, however...

> What happens when there are events in a TD but that TD follows a profile that does not constrain events (like baseline)?

Note that in general conforming with a profile does not prevent a TD from including additional operations outside of the scope of the profile or additional protocol bindings which do not conform with the profile, however...

> In other words, should an event in a baseline profile TD use API key security scheme?

When you say "should" do you mean "could"?

The security section actually only says that Consumers MUST support a particular set of security schemes. There's technically nothing to stop a Thing using another security scheme, but any affordances which require that security scheme are not guaranteed to be consumable.

However, it _does_ currently limit security metadata to the top level of a Thing (I have a [PR](https://github.com/w3c/wot-profile/pull/293) that proposes removing this constraint because it prevents combinations of different profiles and protocol bindings). My understanding from the TD specification is that if multiple security schemes are defined at the top level they _all_ have to be satisfied.

In practice I think this means that if you used a security scheme which isn't required to be supported by Consumers, the whole Thing would become unusable.

I suspect that this is not the intended effect of the security constraints, and what is actually meant is that Things MUST NOT use security schemes other than those listed. Currently it only says that they MAY use the security schemes listed.

In short, I think the answer to your question is that technically yes it could currently use that security scheme, but then the whole Thing would require that security scheme to be satisfied, which may render it unusable by some Consumers.

**Comment by lu-zero** [comment]
We might reconsider for profile 2.0 if we want to have constraining profiles and how they compose. Right now profiles signal expected capabilities.
